cartographer_ros/cartographer_ros/dev/trajectory_comparison_main.cc

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "cartographer/common/math.h"
+#include "cartographer/io/proto_stream.h"
+#include "cartographer/mapping/proto/pose_graph.pb.h"
+#include "cartographer/mapping/proto/trajectory_builder_options.pb.h"
+#include "cartographer/transform/transform_interpolation_buffer.h"
+#include "cartographer_ros/time_conversion.h"
+#include "gflags/gflags.h"
+#include "glog/logging.h"
+#include "cartographer_ros/msg_conversion.h"
+#include "ros/ros.h"
+#include "ros/time.h"
+#include "rosbag/bag.h"
+#include "rosbag/view.h"
+#include "tf2_eigen/tf2_eigen.h"
+#include "tf2_msgs/TFMessage.h"
+
+DEFINE_string(bag_filename, "",
+              "Bags to process, must be in the same order as the trajectories "
+              "in 'pose_graph_filename'.");
+DEFINE_string(pbstream_filename, "",
+              "Proto stream file containing the pose graph.");
+
+namespace cartographer_ros {
+namespace {
+
+double FractionSmallerThan(const std::vector<double>& v, double x) {
+  std::size_t i = 0;
+  for (double value : v) {
```

Comment:
Since you have already included <algorithm>, you might want to replace it with one line:
 `return  std::count_if(v.begin(), v.end(), [=](double value){return value < x;}) / v.size();`.

Reply:
Done